### ISSUE TITLE
Code release

### ISSUE THREAD

**Issue by andresprados** [andresprados/SPIGA#1]

**Comment by Shubhamai** [comment]
Amazing! Looking forward to it!

**Comment by andresprados** [comment]
Sorry for the delay, we hope to have it ready before the start of the conference (21/11/2022).

**Comment by Shubhamai** [comment]
Thanks a lot for the update :)

**Comment by Oguzhanercan** [comment]
How can I run the  inference code?

**Comment by nishitanand** [comment]
Hello, thank you for the amazing work. When will the training code be released? And how can I run the inference code? I tried but wasn't able to. The code doesn't seem complete. I tried running framework.py, config.py and pretreatment.py in the inference folder

**Comment by andresprados** [comment]
Hi, the demo with an inference example is expected to be released next week (01/02/2023). In the meantime, the readme shows how to generate results from different datasets in the [evaluation](https://github.com/andresprados/SPIGA#evaluation) section.

**Comment by jinfagang** [comment]
@andresprados Hello, does the 3d pose estimated also sota compare with previously methods? How's the accuracy and speed tradeoff? Can it in realtime on CPU?

**Comment by andresprados** [comment]
> Hi, the demo with an inference example is expected to be released next week (01/02/2023). In the meantime, the readme shows how to generate results from different datasets in the [evaluation](https://github.com/andresprados/SPIGA#evaluation) section.

Inference framework and image colab demo are already released!
Next week I will add a real-time video demo that will include:
* Face detection.
* Face tracking.
* Face alignment.
* Headpose estimation.
* Python package.

I share with you the preview:
<img src="https://user-images.githubusercontent.com/32907030/216211726-7c3b8d21-4bb2-4795-9864-9ed2ff5d4ed5.png" width="480">

**Comment by Sparno1179** [comment]
Hi, @andresprados 

Thank you for sharing your wonderful project.
I'm very impressed with your work.

I have noticed that the training code has not yet been released, and I was wondering if you could kindly let me know when it is scheduled to be made available. I am very keen to explore and experiment with SPIGA, and having access to the training code would be helpful.

Thank you.

**Comment by andresprados** [comment]
Hi @Sparno1179, thank you for the support.

At the moment, the training code is integrated into a larger framework supported by our lab. I believe that in the next 2-3 months I should be able to create a simple training process for SPIGA. In the meantime, check out the paper [supplementary](https://bmvc2022.mpi-inf.mpg.de/0155_supp.zip), a detailed explanation of the training process as well as the hyperparameters used can be found there. Also take a look at the dataloaders in the repository, we have included all the data augmentors used as well as the configuration set during the training.

Best,
Andrés

**Comment by Kaktusava** [comment]
Hi @andresprados, thank you for your amazing work!

Do you have any updates on the code release for training? Or maybe you can post the training process as it currently is?

**Comment by Xiejiahao233** [comment]
Hi @andresprados, thank you for your excellent work, I am very interested in this. 
When will the training code be released? 
Or I see you said that the training code had been integrated into the framework of your lab. Can you provide a link? 
Thank you.
